File: 이것이 코딩테스트다/05.이진 탐색/4.부품찾기.js
const binarySearch = (array, target, start, end) => {
  while (start <= end) {
    let mid = Math.floor((start + end) / 2);
    if (array[mid] === target) {
      return mid;
    } else if (array[mid] < target) {
      start = mid + 1;
    } else {
      end = mid - 1;
    }
  }
  return Infinity;
};
const texts = [5, [8, 3, 7, 9, 2], 3, [5, 7, 9]];
const n = texts.shift();
const data = texts.shift();
data.sort((a, b) => a - b);
const m = texts.shift();
const check = texts.pop();

for (const c of check) {
  if (binarySearch(data, c, 0, n) === Infinity) {
    console.log("no");
  } else {
    console.log("yes");
  }
}

File: 이것이 코딩테스트다/05.이진 탐색/4.부품찾기.py
def binary_search (array, target, start, end):
  while start <= end:
    mid = (start + end) // 2

    if target == array[mid]:
      return mid
    elif target < array[mid]:
      end = mid- 1
    else:
      start = mid+ 1
  return None

n = int(input())
data = list(map(int, input().split()))
data.sort()
m = int(input())
check = list(map(int, input().split()))

for c in check:
  if binary_search(data, c, 0, n) == None:
    print("no", end=" ")
  else:
    print("yes", end= " ")
